src/Octokit.Webhooks/WebhookEventProcessor.cs

```diff
@@ -156,6 +156,7 @@ headers.Event switch
             WebhookEventType.DependabotAlert => JsonSerializer.Deserialize<DependabotAlertEvent>(body)!,
             WebhookEventType.DeployKey => JsonSerializer.Deserialize<DeployKeyEvent>(body)!,
             WebhookEventType.Deployment => JsonSerializer.Deserialize<DeploymentEvent>(body)!,
+            WebhookEventType.DeploymentProtectionRule => JsonSerializer.Deserialize<DeploymentProtectionRuleEvent>(body)!,
```

Comment:
This also needs to be added above in `ProcessWebhookAsync`

Reply:
Cool, I'll fix that tomorrow and also look at adding a test that would have flushed that out locally.

Reply:
I've added this, though I haven't been able to get the test working as I _think_ I've hit a bug in Moq that means I can get protected mocks to verify correctly. I'm still investigating whether that's the case (rather than me being an idiot) - if it is a bug I'll post it back here for reference and if that gets fixed I can add the test later, if it's being being stupid then I'll either fix it and push it here, or add it post-merge.

Reply:
Have raised an issue with Moq asking if there's a way to get it working here: https://github.com/moq/moq4/issues/1339

Reply:
Thanks for the investigation :heart:

It would be a great addition to verify the entire flow through this class, but I think the current tests are good enough. I'm mostly worried about being able to deserialize a payload.

Reply:
I opened an issue to track this: https://github.com/octokit/webhooks.net/issues/249. Let me know if I've misunderstood anything